modules/common/Riot.py

```diff
@@ -53,30 +53,37 @@ def run_riot(self, owl_file, dir_output, json_file, owl_jq):
         # Set JVM memory limits
         run_env = os.environ.copy()
         run_env["_JAVA_OPTIONS"] = "-Xms4096m -Xmx8192m"
+
         try:
-            with open(path_output, "wb") as json_output, \
-                    subprocess.Popen([self.riot_cmd, "--output", "JSON-LD", owl_file], env=run_env,
-                                     stdout=subprocess.PIPE,
-                                     stderr=subprocess.PIPE) as riot_process, \
-                    subprocess.Popen([self.jq_cmd, "-r", owl_jq], env=run_env,
-                                     stdin=riot_process.stdout,
-                                     stdout=subprocess.PIPE,
-                                     stderr=subprocess.PIPE) as jq_process:
-                json_output.write(jq_process.stdout.read())
-                logger.error(f"RIOT stderr -> '{riot_process.stderr.read()}'")
-                logger.error(f"JQ stderr -> '{jq_process.stderr.read()}'")
-        except EnvironmentError as e:
-            msg_err = "When running RIOT for OWL file '{}', "
-            "with destination path '{}' and JQ filter '{}', "
-            "the following error occurred: '{}'".format(owl_file, path_output, owl_jq, e)
-            logger.error(msg_err)
-            raise RiotException(msg_err)
-        except Exception as e:
-            msg_err = "When running RIOT for OWL file '{}', "
-            "with destination path '{}' and JQ filter '{}', "
-            "the following error occurred: '{}'".format(owl_file, path_output, owl_jq, e)
-            logger.error(msg_err)
-            raise RiotException(msg_err)
+            logger.info(f"running Riot on {owl_file}")
+            riot_output = subprocess.run([self.riot_cmd, "--output", "JSON-LD", owl_file], env=run_env, capture_output=True)
+            riot_output.check_returncode()               
+                                
+            # jq_output = subprocess.run([self.jq_cmd, "-r", owl_jq], input=riot_output.stdout, 
+            #                 stdout=open(path_output, "wb"), stderr=subprocess.PIPE)
+            logger.info(f"running JQ on Riot output...")
+            jq_output = subprocess.run([self.jq_cmd, "-r", owl_jq], input=riot_output.stdout, 
+                                    capture_output= True)
+            jq_output.check_returncode()
+            with open(path_output, "wb") as json_output:
```

Comment:
Yes, I wrote it in this way because I wanted to use check_returncode(). 
Can change it to:
` subprocess.run([self.jq_cmd, "-r", owl_jq], input=subprocess.run([self.riot_cmd, "--output", "JSON-LD", owl_file], env=run_env, capture_output=True).stdout, 
                                    stdout=open(path_output, "wb"))`